README.md
# Univalcord

This is an simple app to load websites lol

## Options

In this section will you see how to add websites

Base folder: `C:\univalcord`

File name: `titlebar.json`

```json
[
  {
    "name": "Discord",
    "link": "https://discord.com/app",
    "titlebarColor": "#24292e",
    "stylesheet": {
      "source": "external",
      "file": "https://raw.githubusercontent.com/Gibbu/BetterDiscord-Themes/master/Slate/base.css"
    },
    "javascript": {
      "source": "internal",
      "file": "discord.js"
    }
  },
  {
    "name": "YouTube",
    "link": "https://www.youtube.com",
    "titlebarColor": "#212121",
    "stylesheet": {
      "source": "internal",
      "file": "youtube.css"
    },
    "javascript": {
      "source": "internal",
      "file": "youtube.js"
    }
  },
  {
    "name": "WhatsApp",
    "link": "https://web.whatsapp.com",
    "titlebarColor": "#1c2026",
    "stylesheet": {
      "source": "internal",
      "file": "whatsapp.css"
    },
    "javascript": {
      "source": "internal",
      "file": "whatsapp.js"
    }
  }
]
```

File name: `settings.json`

```json
{
  "titlebarColor": "#272727",
  "title": "Gay Simulator",
  "startUrl": "https://venge.io"
}
```


CHANGELOG.md
# 3.0.0

- Imporved file size
- 

src/main.ts
import { ElectronBlocker, fullLists, Request } from '@cliqz/adblocker-electron'
import { app, BrowserWindow } from 'electron'
import { Menu, MenuItem } from 'electron'
import { dialog } from 'electron'
import { ipcMain } from 'electron'
import { remote } from 'electron'
import fetch = require('cross-fetch') // required 'fetch'
import fs = require('fs')
import path = require('path')

function createWindow() {
  // Create the browser window.
  const mainWindow = new BrowserWindow({
    width: 1200,
    height: 800,
    frame: false,
    backgroundColor: '#0d1117',
    fullscreenable: true,
    titleBarStyle: "hidden", // add this line
    icon: __dirname + '/icons/icon.ico',
    title: 'Univalcord',
    webPreferences: {
      nodeIntegration: true,
      enableRemoteModule: true,
      nodeIntegrationInSubFrames: true,
      webSecurity: false,
      webviewTag: true,
      preload: path.join(__dirname, '/preload.js'),
      devTools: true
    }
  })
  mainWindow.loadFile(path.join(__dirname, './index.html'))
  mainWindow.setMenuBarVisibility(false);

}

// This method will be called when Electron has finished
// initialization and is ready to create browser windows.
// Some APIs can only be used after this event occurs.
app.on("ready", () => {
  createWindow();

  app.on("activate", function () {
    // On macOS it's common to re-create a window in the app when the
    // dock icon is clicked and there are no other windows open.
    if (BrowserWindow.getAllWindows().length === 0) createWindow();
  });
});

// Quit when all windows are closed, except on macOS. There, it's common
// for applications and their menu bar to stay active until the user quits
// explicitly with Cmd + Q.
app.on("window-all-closed", () => {
  if (process.platform !== "darwin") {
    app.quit();
  }
});

// In this file you can include the rest of your app"s specific main process
// code. You can also put them in separate files and require them here.


src/renderer.ts
// This file is required by the index.html file and will
// be executed in the renderer process for that window.
// No Node.js APIs are available in this process unless
// nodeIntegration is set to true in webPreferences.
// Use preload.js to selectively enable features
// needed in the renderer process.
import { ElectronBlocker, fullLists, Request } from '@cliqz/adblocker-electron'
import { BrowserWindow } from 'electron'
import { session } from 'electron'
import { Menu, MenuItem } from 'electron'
import { remote, shell } from 'electron'
import { dialog } from 'electron'
import fetch = require('cross-fetch') // required 'fetch'
import { Titlebar, Color } from 'custom-electron-titlebar'
import fs = require('fs')
import path = require('path')
const webview = document.getElementById('webview');

//set titlebar
const tilebar = new Titlebar({
    backgroundColor: Color.fromHex('#0d1117'),
    icon: './icons/icon.ico'
});
tilebar.updateTitle('Univalcord');

// set menu or from titlebar.json
const menu = new Menu();
if (!fs.existsSync('C:/univalcord/titlebar.json')) {
    menus()
    menu.append(new MenuItem({
        label: 'Discord',
        submenu: [{
            label: 'This Window',
            click: () => {
                menu_click_make('discord', 'https://discord.com/app', 'Discord', '#202225')
            }
        }]
    }));
    menu.append(new MenuItem({
        label: 'YouTube',
        submenu: [{
            label: 'This Window',
            click: () => {
                menu_click_make('youtube', 'https://www.youtube.com', 'YouTube', '#212121')
            }
        }]
    }));
} else {
    const myUsers = JSON.parse(fs.readFileSync('C:/univalcord/titlebar.json').toString())
    menus()
    const usersByLikes = myUsers.map((item: { name: any; css: any; link: any; titlebarColor: any }) => {
        const container = menu.append(new MenuItem({
            label: item.name,
            submenu: [{
                label: 'This Window',
                click: () => {
                    menu_click_make(item.css, item.link, item.name, item.titlebarColor)
                }
            }]
        }));
        return container;
    })
    console.log(usersByLikes);
}
tilebar.updateMenu(menu);

// ad blocker

// function
function menu_click_make(fileName: string, url: string, name: string, colorTitlrbar: any) {
    webview.setAttribute('src', url)
    webview.addEventListener('dom-ready', function () {
        if (!fs.existsSync('C:/univalcord/' + fileName + '.css')) {
            console.log('No ' + fileName + '.css found')
        } else {
            webview.insertCSS(fs.readFileSync('C:/univalcord/' + fileName + '.css').toString())
        }
        if (!fs.existsSync('C:/univalcord/' + fileName + '.js')) {
            console.log('No ' + fileName + '.js found')
        } else {
            webview.executeJavaScript(fs.readFileSync('C:/univalcord/' + fileName + '.js').toString())
        }
        tilebar.updateTitle(name);
        tilebar.updateBackground(colorTitlrbar);
    });
}

// menu
function menus() {
    menu.append(new MenuItem({
        label: 'View',
        submenu: [{
            label: 'DevTools',
            click: () => {
                if (!fs.existsSync('C:/univalcord/enableDevTools')) {
                    alert('DevTools has been disabled by default')
                } else {
                    if (webview.isDevToolsOpened()) {
                        webview.closeDevTools()
                    } else {
                        webview.openDevTools()
                    }
                }
            }
        },
        {
            label: 'New Window',
            click: () => {
                createwindow()
            }
        }]
    }));
}

// creates new window whits gets called
async function createwindow() {
    const newWindow = new BrowserWindow({
        width: 1200,
        height: 800,
        frame: false,
        backgroundColor: '#0d1117',
        fullscreenable: true,
        titleBarStyle: "hidden", // add this line
        icon: __dirname + '/icons/icon.ico',
        title: 'Univalcord',
        webPreferences: {
            nodeIntegration: true,
            enableRemoteModule: true,
            nodeIntegrationInSubFrames: true,
            webSecurity: false,
            webviewTag: true,
            preload: path.join(__dirname, '/preload.js'),
            devTools: false
        }
    })
    newWindow.loadFile(path.join(__dirname, './index.html'))
    newWindow.setMenuBarVisibility(false);
}